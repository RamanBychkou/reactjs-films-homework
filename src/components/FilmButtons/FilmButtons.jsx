import React from 'react';
import style from './FilmButtons.scss';

export const showMainFilmOverwie = () => {
  document.querySelector('#mainFilmOverwie').style.maxHeight = '100%';
}

const FilmButtons = () => (
  <div className={style.btnContainer}>
    <button id="watch" type="button" className={style.watch}>Watch Now</button>
    <button id="info" type="button" className={style.viewInfo} onClick={showMainFilmOverwie}>View Info</button>
  </div>
);

export default FilmButtons;
