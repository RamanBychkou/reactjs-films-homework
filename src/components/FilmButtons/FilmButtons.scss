.watch {
  background-color: #09ade3;
  border: 1px solid #09ade3;
  margin-right: 1.3rem;
}

.viewInfo {
  background: none;
  border: 1px solid #fff;
}

.watch,
.viewInfo {
  border-radius: 3px;
  cursor: pointer;
  outline: none;
  color: #fff;
  text-align: center;
  vertical-align: middle;
  width: 6.2rem;
  height: 1.6rem;
}

.btnContainer {
  margin: 2.2rem 3.5rem 0.2rem 5rem;
  display: flex;
  justify-content: space-between;
}
