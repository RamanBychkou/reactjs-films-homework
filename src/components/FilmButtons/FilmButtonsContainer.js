import { connect } from 'react-redux';
import requestsFilms from '../../utils/requests';
import FilmButtons from './FilmButtons';

const mapStateToProps = state => state;

export const mapStateToDispatch = dispatch => ({
  fetchVideo: requestsFilms.fetchVideo.bind(this, dispatch),
});

export default connect(
  mapStateToProps,
  mapStateToDispatch,
)(FilmButtons);
