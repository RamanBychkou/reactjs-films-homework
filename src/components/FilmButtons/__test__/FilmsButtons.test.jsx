import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import FilmButtons from '../index';
import style from '../FilmButtons.scss';
import { showMainFilmOverwie } from '../FilmButtons';


test('Signature renders correctly', () => {
  const renderer = new ShallowRenderer();
  const initialState = {
    movies: {
      page: 0,
      results: [],
      mainMovie: {},
      currentVideo: null,
    },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);
  const result = renderer.render(
    <Provider store={store}><FilmButtons className={style.name} /></Provider>,
  );
  expect(result).toMatchSnapshot();
});

test('Signature renders correctly', () => {
  const renderer = TestRenderer;
  const initialState = {
    movies: {
      page: 0,
      results: [],
      mainMovie: {},
      currentVideo: null,
    },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);
  const result = renderer.create(
    <Provider store={store}><FilmButtons className={style.name} /></Provider>,
  );
  const node = result.root.findByProps({ id: 'info' });
  expect(ReactTestUtils.Simulate.click(node)).toMatchSnapshot();
});
