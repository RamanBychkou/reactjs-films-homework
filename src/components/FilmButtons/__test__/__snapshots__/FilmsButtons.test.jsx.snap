// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signature renders correctly 1`] = `
<div
  className="btnContainer"
>
  <button
    className="watch"
    id="watch"
    type="button"
  >
    Watch Now
  </button>
  <button
    className="viewInfo"
    id="info"
    onClick={[Function]}
    type="button"
  >
    View Info
  </button>
</div>
`;
